finality-aleph/src/metrics.rs

```diff
@@ -1,34 +1,37 @@
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::{collections::HashMap, time::Instant};
+
 use log::trace;
+use lru::LruCache;
 use parking_lot::Mutex;
 use prometheus_endpoint::{register, Gauge, PrometheusError, Registry, U64};
 use sc_service::Arc;
-use sp_runtime::traits::Header;
-use std::{collections::HashMap, time::Instant};
 
-#[derive(Clone)]
-struct Inner<H: Header> {
+// How many entries (block hash + timestamp) we keep in memory per one checkpoint type.
+const MAX_BLOCKS_PER_CHECKPOINT: usize = 20;
```

Comment:
Now I think it would be good to state in a comment a rough calculation of how much memory this will use depending on `MAX_BLOCKS_PER_CHECKPOINT` -- of course we are talking about order of magnitude, not counting each single byte.

Reply:
good point, adapted